main.js

```diff
@@ -1,29 +1,40 @@
 const { join } = require('node:path');
 const { readFileSync } = require('node:fs');
 
-const lockfile = join(process.env.GITHUB_WORKSPACE, 'package-lock.json');
-const matcher = /.+\/(?<repo>[a-z-]+\/[a-z-]+)\.git#(?<hash>[0-9a-f]+)/;
+const lockfile = join(process.env.GITHUB_WORKSPACE ?? process.cwd(), 'package-lock.json');
+const matcher = /.+\/(?<repo>[\w-]+\/[\w-]+)\.git#(?<hash>[0-9a-f]+)/;
 
-function getBody() {
+/** @param {import('@actions/core')} core */
+function getBody(core) {
   const { packages } = JSON.parse(readFileSync(lockfile, 'utf8'));
   return Object.entries(packages).slice(1).map(([key, value]) => {
     if (value.resolved.startsWith('git')) {
-      const { repo, hash } = matcher.exec(value.resolved).groups;
-      return `- [${repo}@\`${hash}\`](https://github.com/${repo}/commit/${hash})`;
+      const match = matcher.exec(value.resolved)
+      if (match) {
+        // @ts-ignore
+        const { repo, hash } = match.groups;
+        return `- [${repo}@\`${hash}\`](https://github.com/${repo}/commit/${hash})`;
+      } else {
+        core.warning(`
+          Could not parse ${value.resolved}.
```

Comment:
This needs to be a one-liner in order to be shown in log. But "could not parse" might be enough